arrow-libs/integrations/arrow-raise-ktor-server/src/commonMain/kotlin/arrow/raise/ktor/server/response/respondOrRaise.kt

```diff
@@ -0,0 +1,62 @@
+package arrow.raise.ktor.server.response
+
+import arrow.core.raise.Raise
+import arrow.core.raise.RaiseDSL
+import arrow.core.raise.fold
+import arrow.core.raise.withError
+import arrow.raise.ktor.server.response.Response.Companion.Response
+import io.ktor.http.*
+import io.ktor.http.content.*
+import io.ktor.server.application.*
+import io.ktor.server.response.*
+import io.ktor.util.reflect.*
+
+public suspend inline fun <reified TResponse> ApplicationCall.respondOrRaise(
+  statusCode: HttpStatusCode? = null,
+  body: suspend Raise<Response>.() -> TResponse,
```

Comment:
@serras with the comment around [ambiguity](https://github.com/arrow-kt/arrow/pull/3646#issuecomment-3056549399) in #3646 is this better defined as follows given the `verbOrRaise` helpers use contextual Raise (to be consistent across the whole module)?

```suggestion
  body: suspend context(Raise<Response>) () -> TResponse,
```

Reply:
Yes!